sdk/src/main/java/io/dapr/client/domain/StateKeyValue.java

```diff
@@ -4,25 +4,57 @@
  */
 package io.dapr.client.domain;
 
+/**
+ * This class reprent what a State is
+ * @param <T>
+ */
 public class StateKeyValue<T> {
+  /**
+   * The value of the state
+   */
   private final T value;
+  /**
+   * The key of the state
+   */
   private final String key;
+  /**
+   * The ETag to be used
+   * For REDIS ONLY this must be an integer
```

Comment:
Make it more generic, saying that in some state stores (like Redis), this must be an integer.